### PR TITLE
Fixed type error in apply-decorators.ts

### PR DESCRIPTION
**Pull request by nikelborm**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Throws typescript error
```
Argument of type 'string | symbol | undefined' is not assignable to parameter of type 'string | symbol'.
  Type 'undefined' is not assignable to type 'string | symbol'
```

Here:
Screenshot from 2024-01-26 04-06-03

Because [typescript definitions of Decorator types](https://github.com/microsoft/TypeScript/blob/main/src/lib/decorators.legacy.d.ts) are as follows:
```typescript
declare type ClassDecorator = <TFunction extends Function>(target: TFunction) => TFunction | void;
declare type PropertyDecorator = (target: Object, propertyKey: string | symbol) => void;
declare type MethodDecorator = <T>(target: Object, propertyKey: string | symbol, descriptor: TypedPropertyDescriptor<T>) => TypedPropertyDescriptor<T> | void;
declare type ParameterDecorator = (target: Object, propertyKey: string | symbol | undefined, parameterIndex: number) => void;
```

## What is the new behavior?
Doesn't throw an error
Screenshot from 2024-01-26 03-42-21

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by nikelborm** [comment]
Apply-decorators were introduced here https://github.com/nestjs/nest/pull/3113

File: packages/common/decorators/core/apply-decorators.ts [hunk 1]
```diff
@@ -14,16 +14,28 @@ export function applyDecorators(
...
     for (const decorator of decorators) {
-      if (target instanceof Function && !descriptor) {
+      if (target instanceof Function && descriptor === undefined && propertyKey === undefined) {
         (decorator as ClassDecorator)(target);
         continue;
       }
-      (decorator as MethodDecorator | PropertyDecorator)(
+
+      if (propertyKey === undefined)
+        throw new Error('Parameter decorators are not supported in applyDecorators function');
```
Comment (kamilmysliwiec) [inline packages/common/decorators/core/apply-decorators.ts:25]: nit: wrap with {}

Comment (nikelborm) [reply packages/common/decorators/core/apply-decorators.ts]: done

**Comment by kamilmysliwiec** [comment]
https://github.com/nestjs/nest/commit/212d07fcbf09c973270e21c6d4cea3c47f6265c5

**Pull request closed without merging**